libs/clients/regulations-admin/src/lib/RegulationsAdminClientService.ts

```diff
@@ -1,62 +1,194 @@
-import { RESTDataSource, RequestOptions } from 'apollo-datasource-rest'
-import { DataSourceConfig } from 'apollo-datasource'
 import {
+  DraftImpact,
+  DraftRegulationChange,
   RegulationDraft,
-  ShippedSummary,
-  TaskListType,
 } from '@island.is/regulations/admin'
+import {
+  DraftAuthorApi,
+  DraftRegulationCancelApi,
+  DraftRegulationChangeApi,
+  DraftRegulationsApi,
+} from '../../gen/fetch/apis'
+import { Auth, AuthMiddleware } from '@island.is/auth-nest-tools'
 import { RegulationsAdminClientConfig } from './RegulationsAdminClientConfig'
 import { ConfigType } from '@nestjs/config'
+import {
+  CreateDraftRegulationCancelDto,
+  CreateDraftRegulationChangeDto,
+  CreateDraftRegulationDto,
+  DraftRegulationCancelModel,
+  UpdateDraftRegulationCancelDto,
+  UpdateDraftRegulationChangeDto,
+  UpdateDraftRegulationDto,
+} from '../../gen/fetch/models'
 
-export class RegulationsAdminClientService extends RESTDataSource {
+export class RegulationsAdminClientService {
   constructor(
     @Inject(RegulationsAdminClientConfig.KEY)
     private readonly config: ConfigType<typeof RegulationsAdminClientConfig>,
-  ) {
-    super()
-    this.baseURL = `${this.config.baseApiUrl}`
-    this.initialize({} as DataSourceConfig<any>)
-  }
+    private draftAuthorApi: DraftAuthorApi,
+    private draftRegulationCancelApi: DraftRegulationCancelApi,
+    private draftRegulationChangeApi: DraftRegulationChangeApi,
+    private draftRegulationsApi: DraftRegulationsApi,
+  ) {}
 
-  willSendRequest(request: RequestOptions) {
-    this.memoizedResults.clear()
-    request.headers.set('Content-Type', 'application/json')
+  draftAuthorApiWithAuth(auth: Auth) {
+    return this.draftAuthorApi.withMiddleware(new AuthMiddleware(auth))
   }
 
-  async getDraftRegulations(authorization: string, page?: number) {
-    return await this.get<TaskListType>(
-      '/draft_regulations',
-      {
-        page,
-      },
-      {
-        headers: { authorization },
-      },
+  draftRegulationCancelApiWithAuth(auth: Auth) {
+    return this.draftRegulationCancelApi.withMiddleware(
+      new AuthMiddleware(auth),
     )
   }
 
-  async getShippedRegulations(authorization: string) {
-    return await this.get<ShippedSummary[]>(
-      `/draft_regulations_shipped`,
-      {},
-      {
-        headers: { authorization },
-      },
+  draftRegulationChangeApiWithAuth(auth: Auth) {
+    return this.draftRegulationChangeApi.withMiddleware(
+      new AuthMiddleware(auth),
     )
   }
 
+  draftRegulationsApiWithAuth(auth: Auth) {
+    return this.draftRegulationsApi.withMiddleware(new AuthMiddleware(auth))
+  }
+
+  async getDraftRegulations(auth: Auth, page?: number) {
+    const allDrafts = await this.draftRegulationsApiWithAuth(
+      auth,
+    ).draftRegulationControllerGetAll({
+      page: page ?? 0,
+    })
+    return allDrafts
+  }
+
+  async getShippedRegulations(auth: Auth) {
+    return await this.draftRegulationsApiWithAuth(
```

Comment:
Unnecessary `await` and then the `async` can be removed from the function definition. Should just simplify this for all functions in this service that are doing `return await ...`
```suggestion
    return this.draftRegulationsApiWithAuth(
```